webapp/src/components/BidAssetPage/BidAssetForm.js

```diff
@@ -0,0 +1,184 @@
+import { t } from '@dapps/modules/translation/utils'
+
+import { assetType, bidType } from 'components/types'
+import { preventDefault, formatDate, formatMana } from 'lib/utils'
+import { isParcel } from 'shared/parcel'
+import {
+  DEFAULT_DAY_INTERVAL,
+  MINIMUM_DAY_INTERVAL,
+  MAXIMUM_BID_DAY_INTERVAL,
+  MINIMUM_ASSET_PRICE
+} from 'shared/listing'
+import TxStatus from 'components/TxStatus'
+
+const INPUT_FORMAT = 'YYYY-MM-DD'
+
+export default class BidAssetForm extends React.PureComponent {
+  static propTypes = {
+    asset: assetType,
+    assetName: PropTypes.string,
+    bid: bidType,
+    isTxIdle: PropTypes.bool,
+    isDisabled: PropTypes.bool,
+    onBid: PropTypes.func.isRequired,
+    onCancel: PropTypes.func.isRequired
+  }
+
+  constructor(props) {
+    super(props)
+    const { bid } = props
+
+    this.state = {
+      price: bid ? bid.price : '',
+      expiresAt: bid
+        ? dateFnsFormat(parseInt(bid.expires_at, 10), INPUT_FORMAT)
+        : this.formatFutureDate(DEFAULT_DAY_INTERVAL),
+      formErrors: []
+    }
+  }
+
+  handlePriceChange = e => {
+    // Dots and commas are not allowed
+    const price = e.currentTarget.value.replace(/\.|,/g, '')
+
+    this.setState({
+      price,
+      formErrors: []
+    })
+  }
+
+  handleExpiresAtChange = e => {
+    this.setState({
+      expiresAt: e.currentTarget.value,
+      formErrors: []
+    })
+  }
+
+  handleClearFormErrors = () => {
+    this.setState({ formErrors: [] })
+  }
+
+  handlePublish = () => {
+    const { asset, onBid } = this.props
+    const { price, expiresAt } = this.state
+
+    const formErrors = []
+
+    const today = new Date()
+    today.setHours(0, 0, 0, 0)
+
+    if (differenceInDays(expiresAt, today) < MINIMUM_DAY_INTERVAL) {
+      formErrors.push(
+        t('asset_bid.errors.minimum_expiration', {
+          date: this.formatFutureDate(MINIMUM_DAY_INTERVAL + 1)
+        })
+      )
+    }
+
+    if (differenceInDays(expiresAt, today) > MAXIMUM_BID_DAY_INTERVAL) {
+      formErrors.push(
+        t('asset_bid.errors.maximum_expiration', {
+          date: this.formatFutureDate(MAXIMUM_BID_DAY_INTERVAL)
+        })
+      )
+    }
+
+    if (price < MINIMUM_ASSET_PRICE) {
+      formErrors.push(
+        t('asset_bid.errors.minimum_land_price', {
+          value: formatMana(MINIMUM_ASSET_PRICE, ''),
+          asset_name: isParcel(asset) ? t('name.parcel') : t('name.estate')
+        })
+      )
+    }
+
+    if (price > Number.MAX_SAFE_INTEGER) {
+      formErrors.push(
+        t('asset_bid.errors.maximum_land_price', {
+          value: formatMana(Number.MAX_SAFE_INTEGER, ''),
+          asset_name: isParcel(asset) ? t('name.parcel') : t('name.estate')
+        })
+      )
+    }
+
+    if (formErrors.length === 0) {
+      onBid({
+        asset_id: asset.id,
+        expires_at: this.toUTCTimestamp(expiresAt),
+        price: parseInt(price)
+      })
+    } else {
+      this.setState({ formErrors })
+    }
+  }
```

Comment:
Same goes for these error checks, they're practically the same as PublishAssetForm. This abstraction is not **that** easy, so it can be added later but we should add an issue!